tutorials/tools/Text_Normalization_Tutorial.ipynb

```diff
@@ -0,0 +1,472 @@
+        "\n",
+        "We additionally added the class `WHITELIST` for all whitelisted tokens whose verbalizations directly looked up from `$TOOLS_DATA_DIR/whitelist.tsv`. To extend the list simply add further key value pairs to the file.\n",
+        "\n",
+        "\n",
+        "We will create a simple example file extracted from the Google Text normalization dataset to show how evaluation works:\n",
+        "\n",
+        "\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "u4zjeVVv-UXR"
+      },
+      "source": [
+        "eval_input_data =  \"\"\"LETTERS\\tA & E\\ta and e\r\n",
+        "PUNCT\\t.\\tsil\r\n",
+        "PLAIN\\tRetrieved\\t<self>\r\n",
+        "DATE\\t2006-08-05\\tthe fifth of august two thousand six\r\n",
+        "PUNCT\\t.\\tsil\r\n",
+        "<eos>\\t<eos>\r\n",
+        "PLAIN\\tDownloaded\\t<self>\r\n",
+        "PLAIN\\ton\\t<self>\r\n",
+        "DATE\\t7 August 2007\\tthe seventh of august two thousand seven\r\n",
+        "PUNCT\\t.\\tsil\"\"\"\r\n",
+        "eval_text_file_path = f\"{DATA_DIR}/00001-of-00100\"\r\n",
+        "with open(eval_text_file_path, 'w') as fp:\r\n",
+        "  fp.write(eval_input_data)\r\n",
+        "! cat $eval_text_file_path\r\n"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "kmDTCuTLH7pm"
+      },
+      "source": [
+        "# Prediction\n",
+        "Here we will show `$TOOLS_DIR/run_predict.py` step by step\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "6R7OKAsYH9p0"
+      },
+      "source": [
+        "\r\n",
+        "import tools.text_normalization.verbalizer as verbalizer \r\n",
+        "import tools.text_normalization.tagger as tagger \r\n",
+        "import tools.text_normalization.normalize as normalize\r\n",
+        "from tools.text_normalization.run_predict import load_file, write_file"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "O7VpoBMqocQQ"
+      },
+      "source": [
+        "data = load_file(f\"{DATA_DIR}/{text_file}\")\r\n",
+        "print(len(data), \"sentences\") "
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "3etPl6smpYrE"
+      },
+      "source": [
+        "If you want to see how things were normalized, turn on `verbose=True` flag\r\n",
+        "\r\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "f-AAXW8ZovZo"
+      },
+      "source": [
+        "# normalized_sentences = normalize.normalize_nemo(data, verbose=True)\r\n",
+        "normalized_sentences = normalize.normalize_nemo(data, verbose=False)"
```

Comment:
Do you want to leave the comment? Is the log stored anywhere?

Reply:
yes i want to leave the comment its not stored anywhere